sdcm/nemesis.py

```diff
@@ -2398,6 +2399,165 @@ def disrupt_memory_stress(self):
         self.target_node.remoter.run(
             "stress-ng --vm-bytes $(awk '/MemTotal/{printf \"%d\\n\", $2 * 0.9;}' < /proc/meminfo)k --vm-keep -m 1 -t 100")
 
+    def disrupt_toggle_cdc_feature_properties_on_table(self):
+        """Manupulate cdc feature
+
+        randomly choose table and toggle cdc feature
+        with radomly generated cdc properties
+        """
+        self._set_current_disruption("ToggleCDCProperties")
+
+        if not self.tester.params.get("cdc_nemesis_enabled"):
+            self.log.warning("CDCStressor couldn't be run, executing cdc nemesis is disabled")
+            UnsupportedNemesis("CDCStressor couldn't be run, executing cdc nemesis is disabled")
+            return
+
+        def _generate_cdc_properties(cdc_enabled: bool) -> Tuple[str, str, str, int]:
+            """Generate random set of cdc properties
+
+            :param cdc_enabled: Was cdc enabled on table (True) or not (False)
+            :type cdc_enabled: bool
+            """
+            cdc_enabled = random.choice([True, False]) if cdc_enabled else True
+            preimage = random.choice([True, False]),
+            postimage = random.choice([True, False]),
+            ttl = random.randint(300, 3600)
```

Comment:
I think this should use "toggle" approach and not random true/false.
It means that if the table had cdc enabled, we can disable it or change preimage/postimage/ttl.
but changing would be flipping.

What we can do to avoid reaching states we don't know, we can sleep / wait for the changes to take effect (how do we know?) and then flip it back to the original value.

This will prevent cases where a job with CDC will hit this nemesis in the very beginning, disable the CDC and won't enable it till it ends.

Reply:
Fixed. if cdc was enabled, we disable it for a while and the enable it back
if cdc was disabled, we enable it choose random settings for preimage and post image, apply new settigns and execute cdc stressor for 5-20 minutes and then return disable cdc.